### DISCUSSION TITLE
Found lockfile missing swc dependencies - npm workspaces

### DISCUSSION THREAD

**Discussion by nickBes** [vercel/next.js#73479] [Help]
### Summary

Hey, we're attempting to create a monorepo using turborepo and npm workspaces. One of our applications is a next js app, and when w'ere trying to run `turbo run build` for this app, we are getting the next error: `Found lockfile is missing swc dependencies, patching...`. After that there's an attempt to install them from the npmjs package registry, which fails as we're working in  closed network with our own registry.

I believe this happens because in npm workspaces there's a single `package-lock.json` file for the root package.json which includes the dependencies for all of the packages in the workspaces. To actually test that, I copied the previous `packege-lock.json` (from the previous repository) and the build finished successfully...

Sometimes I also got the next error (which could actually relate to this [thread](https://github.com/npm/cli/issues/6099)):

```
npm ERR! code ENOWORKSPACES
npm ERR! This command does not support workspaces.
```

### Additional information

```js
Can't bring screenshots / errors
```

**Comment by bestickley** [comment]
@nickBes, did you ever resolve this issue?

**Reply by nickBes** [reply]
Hey, after a while, not sure how, we got different errors which I mentioned in this thread: https://github.com/vercel/next.js/discussions/75408.

Eventually we used the undocumented `NEXT_IGNORE_INCORRECT_LOCKFILE` env var which solved the issue for us (next@^14.2.13)

**Comment by KaloudasDev** [comment]
Hello! This is a known issue with Next.js in npm workspaces environments. Here are the proven solutions:

**Solution 1: Use NEXT_IGNORE_INCORRECT_LOCKFILE (Recommended)**
Set this environment variable to bypass the lockfile validation:
```bash
# In your build command
NEXT_IGNORE_INCORRECT_LOCKFILE=true turbo run build

# Or add to your .env file
NEXT_IGNORE_INCORRECT_LOCKFILE=true
```

**Solution 2: Configure npm Workspace Properly**
Ensure your root `package.json` has correct workspace configuration:
```json
{
  "name": "monorepo-root",
  "workspaces": [
    "apps/*",
    "packages/*"
  ],
  "scripts": {
    "build": "turbo run build"
  }
}
```

**Solution 3: Update Next.js and Dependencies**
Upgrade to the latest Next.js version which has better workspace support:
```bash
npm update next@latest
# Or in your app's package.json
"dependencies": {
  "next": "^14.2.13"
}
```

**Solution 4: Custom Registry Configuration**
If using a private registry, configure npm explicitly:
```bash
# .npmrc in your Next.js app directory
registry=https://your-private-registry.com
always-auth=true
```

**Solution 5: Alternative Build Approach**
For complex workspace setups, consider building outside the workspace context:
```json
{
  "scripts": {
    "build:next": "cd apps/your-next-app && npm run build"
  }
}
```

The `NEXT_IGNORE_INCORRECT_LOCKFILE` environment variable is the most reliable workaround for this specific issue in closed network environments with private registries.
